api/client/client.go

```diff
@@ -190,12 +196,9 @@ func connect(ctx context.Context, cfg Config) (*Client, error) {
 
 			// Connect with dialer provided in config.
 			if cfg.Dialer != nil {
-				syncConnect(func() (*Client, error) {
-					clt, err := dialerConnect(ctx, cfg, cfg.Dialer, tlsConfig)
-					if err != nil {
-						return nil, trace.Wrap(err, "failed to connect using cfg dialer")
-					}
-					return clt, err
+				syncConnect(ctx, dialerConnect, connectParams{
```

Comment:
Just checking my understanding: Packing these values that _used to be_ implicitly captured into the parameter struct forces a copy that fixes the aliasing bug, yes?

Reply:
Correct, sorry for the lack of explanation